src/main/java/mods/railcraft/common/worldgen/Generator.java

```diff
@@ -39,7 +39,7 @@ public final void generate(OreGenEvent.Post event) {
 
     @Override
     public final void generate(Random rand, int chunkX, int chunkZ, World world, IChunkGenerator chunkGenerator, IChunkProvider chunkProvider) {
-        _generate(rand, new BlockPos((chunkX << 4), 0, (chunkZ << 4)), world);
+        _generate(rand, new BlockPos((chunkX << 4) + 8, 0, (chunkZ << 4) + 8), world);
```

Comment:
Does this not counter all your - 8 additions? 😕

Reply:
I am just regulating them so that in that abstract generate method below, i pass in the center pos (+8)

Reply:
Then was it the setBlock changes that actually fixed the problem?

Reply:
No. This also modifies the block lookup range (both the lookup of old blocks and setting of new blocks cause problems)

Before, we had 2 separate issues:
1. Lookup of blocks goes to other chunks and causing cascading world gen
2. Setting of new blocks cause observer updates and loading of adjacent chunks if on chunk border

These two issues are fixed together here.